### PR TITLE
feat(auth): adds X509 workload certificate provider

### PR DESCRIPTION
**Pull request by aeitzman**
See  go/x509-workload-auth-library-design and go/guac-cba-x509-federation-golang.

Adds support for the new certificate provider type and unit tests, but does not add it to the default certificate provider logic yet.

### CODE REVIEW

File: auth/internal/transport/cert/workload_cert.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+		return "", "", err
+	}
+	var config certificateConfig
+	err = json.Unmarshal(byteValue, &config)
+	if err != nil {
+		return "", "", err
+	}
+
+	if config.CertConfigs.Workload == nil {
+		return "", "", errors.New("workload certificate information not found in certificate configuration")
```
Comment (lsirac) [inline auth/internal/transport/cert/workload_cert.go:102]:
```suggestion
		return "", "", errors.New("Workload Identity Federation certificate information not found in the certificate configuration file.")
```

File: auth/internal/transport/cert/workload_cert.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+	}
+
+	certFile := config.CertConfigs.Workload.CertPath
+	keyFile := config.CertConfigs.Workload.KeyPath
+
+	if certFile == "" {
+		err = errors.New("certificate file location could not be found in the certificate configuration")
```
Comment (lsirac) [inline auth/internal/transport/cert/workload_cert.go:109]:
```suggestion
		err = errors.New("The certificate configuration is missing the certificate file location.")
```

File: auth/internal/transport/cert/workload_cert.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	}
+
+	if keyFile == "" {
+		err = errors.New("key file location could not be fouind in the certificate configuration")
```
Comment (lsirac) [inline auth/internal/transport/cert/workload_cert.go:113]:
```suggestion
		err = errors.New("The certificate configuration is missing the key file location.")
```

**Review by lsirac — APPROVED** [review]

File: auth/internal/transport/cert/workload_cert.go (hunk 2 above)
Comment (aeitzman) [reply auth/internal/transport/cert/workload_cert.go:109]: https://go.dev/wiki/CodeReviewComments#error-strings go error messages are not supposed to be capitalized or end with punctuation, will made the wording changes though

File: auth/internal/transport/cert/workload_cert.go (hunk 1 above)
Comment (codyoss) [reply auth/internal/transport/cert/workload_cert.go]: Go errors should begin with a lower case letter and not end in punctuation: https://go.dev/wiki/CodeReviewComments#error-strings

Comment (codyoss) [reply auth/internal/transport/cert/workload_cert.go]: This applies to all open comments

File: auth/internal/transport/cert/workload_cert.go [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+		return "", "", err
+	}
+	var config certificateConfig
+	err = json.Unmarshal(byteValue, &config)
+	if err != nil {
+		return "", "", err
+	}
+
+	if config.CertConfigs.Workload == nil {
+		return "", "", errors.New("Workload Identity Federation certificate information not found in the certificate configuration file")
```
Comment (codyoss) [inline auth/internal/transport/cert/workload_cert.go:102]: nit: can this start with a lower case letter. Can rephrase if needed and you want to capitalize WIF

File: auth/internal/transport/cert/workload_cert.go [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+	}
+
+	certFile := config.CertConfigs.Workload.CertPath
+	keyFile := config.CertConfigs.Workload.KeyPath
+
+	if certFile == "" {
+		err = errors.New("the certificate configuration is missing the certificate file location")
```
Comment (codyoss) [inline auth/internal/transport/cert/workload_cert.go:109]: nit: for this and below avoid trying to make error into a sentencte. Remove the

File: auth/internal/transport/cert/workload_cert.go [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+	}
+
+	if keyFile == "" {
+		err = errors.New("the certificate configuration is missing the key file location")
+	}
+
+	if err != nil {
```
Comment (codyoss) [inline auth/internal/transport/cert/workload_cert.go:116]: This block can be removed, error checked above.

**Review by codyoss — COMMENTED** [review]
mostly just nit, otherwise LGTM

**Review by codyoss — APPROVED** [review]

**Pull request merged**